telemetry/trace-kafka/src/main/scala/io/buoyant/telemetry/KafkaTelemeter.scala

```diff
@@ -0,0 +1,198 @@
+      closeAwaitably(Future({
+        cleanup()
+      }))
+    }
+  }
+}
+
+//Only support thrift for now.
+class KafkaRawZipkinTracer(
+  brokerList: String,
+  numRetries: Int = 3, //this is the kafka default.
+  topic: String = "zipkin",
+  timer: Timer = DefaultTimer.twitter,
+  statsReceiver: StatsReceiver = DefaultStatsReceiver
+) extends RawZipkinTracer(statsReceiver, timer) {
+  private val METADATA_FETCH_TIMEOUT_MS_CONFIG: Int = 3000
+  /* We do not want to wait too long on the broker for the event propagated to the replicas */
+  private val TIMEOUT_MS_CONFIG: Int = 300
+  private val RETRY_BACKOFF_MS_CONFIG: Int = 10000
+  private val RECONNECT_BACKOFF_MS_CONFIG: Int = 10000
+  //Set this to 1 millisecond so that there is enough time for a batch of spans to be pushed to kafka local buffers.
+  private val LINGER_MS_CONFIG = 1
+  private val log = Logger.get(getClass)
+  log.info("Initializing kafka zipkin tracer")
+  private val producer: KafkaProducer[Array[Byte], Array[Byte]] = {
+    val configs: java.util.Map[String, AnyRef] = new java.util.HashMap[String, AnyRef]
+    configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    configs.put(ProducerConfig.METADATA_FETCH_TIMEOUT_CONFIG, METADATA_FETCH_TIMEOUT_MS_CONFIG.asInstanceOf[AnyRef])
+    configs.put(ProducerConfig.TIMEOUT_CONFIG, TIMEOUT_MS_CONFIG.asInstanceOf[AnyRef])
+    configs.put(ProducerConfig.RETRY_BACKOFF_MS_CONFIG, RETRY_BACKOFF_MS_CONFIG.asInstanceOf[AnyRef])
+    configs.put(ProducerConfig.RECONNECT_BACKOFF_MS_CONFIG, RECONNECT_BACKOFF_MS_CONFIG.asInstanceOf[AnyRef])
+    configs.put(ProducerConfig.RETRIES_CONFIG, numRetries.asInstanceOf[AnyRef])
+    configs.put(ProducerConfig.LINGER_MS_CONFIG, LINGER_MS_CONFIG.asInstanceOf[AnyRef])
+    new KafkaProducer[Array[Byte], Array[Byte]](configs, new ByteArraySerializer(), new ByteArraySerializer())
+  }
+
+  //retries configured from the input.
+  private[this] val scopedReceiver = statsReceiver.scope("log_span")
+  private[this] val okCounter = scopedReceiver.counter("ok")
+  private[this] val errorReceiver = scopedReceiver.scope("error")
+
+  private[this] def serializeSpan(span: Span): Array[Byte] = {
+    log.debug("Serializing span data to bytes")
+    val tSerializer = new TSerializer()
+    tSerializer.serialize(span.toThrift)
+  }
+
+  //delegate batching to downstream for simplicity reasons rather than doing buffer pools here.
+  //The one complication it adds is it is hard to do retries. Breaking it into the model if any one stat emission
+  //fails the entire batch is considered to be failed. The batch will only fail if the retries for all the different
+  //stats emitted as part of the batch are exhausted independently of each other
+  // without being able to send it out successfully.
+  // Will try to do early detection. Currently this will enable both
+  //kafka level and higher level retries if the framework supports it.
+  def sendSpans(spans: Seq[Span]): Future[Unit] = {
```

Comment:
If this is part of the RawZipkinTracer interface, we tend to declare it as `override def sendSpans` so that it's clear that it's part of the parent type.